Dockerfile

```diff
@@ -1,71 +1,50 @@
   echo "Europe/London" > /etc/timezone
 
-# build-base: complication tools for bundle
+# build-base: compilation tools for bundle
 # yarn: node package manager
 # postgresql-dev: postgres driver and libraries
-RUN apk add --no-cache build-base yarn postgresql-dev git
+RUN apk add --no-cache build-base yarn postgresql-dev git bash
 
-# Install bundler to run bundle exec
-# This should be the same version as the Gemfile.lock
+# Bundler version should be the same version as what the Gemfile.lock was bundled with
 RUN gem install bundler:2.3.14 --no-document
 
-# Install gems defined in Gemfile
 COPY .ruby-version Gemfile Gemfile.lock /app/
+RUN bundle install --jobs=4 --no-binstubs --no-cache
 
-RUN bundle config set without "development test"
-ARG BUNDLE_FLAGS="--jobs=4 --no-binstubs --no-cache"
-RUN bundle install ${BUNDLE_FLAGS}
-
-# Install node packages defined in package.json, including webpack
 COPY package.json yarn.lock /app/
 RUN yarn install --frozen-lockfile
 
-# Copy all files to /app (except what is defined in .dockerignore)
 COPY . /app/
 
-# Compile assets and run webpack. We set a dummy secret key.
-RUN RAILS_ENV=production bundle exec rails SECRET_KEY_BASE=pickasecuretoken assets:precompile
+RUN bundle exec rake assets:precompile
 
-# Cleanup to save space in the production image
-RUN rm -rf node_modules log tmp && \
-      rm -rf /usr/local/bundle/cache && \
-      rm -rf .env && \
-      find /usr/local/bundle/gems -name "*.c" -delete && \
-      find /usr/local/bundle/gems -name "*.h" -delete && \
-      find /usr/local/bundle/gems -name "*.o" -delete && \
-      find /usr/local/bundle/gems -name "*.html" -delete
+ENV PORT=8080
 
-# Build runtime image
-FROM ruby:3.1.4-alpine as production
+EXPOSE ${PORT}
 
-# The application runs from /app
-WORKDIR /app
+FROM base as development
 
-# Add postgres driver library
-# Add the timezone as it's not configured by default in Alpine
-RUN apk add --update --no-cache libpq tzdata && \
-  cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
-  echo "Europe/London" > /etc/timezone
+# Install gecko driver for Capybara tests
+RUN apk add firefox
+RUN wget https://github.com/mozilla/geckodriver/releases/download/v0.31.0/geckodriver-v0.31.0-linux64.tar.gz \
+  && tar -xvzf geckodriver-v0.31.0-linux64.tar.gz \
+  && rm geckodriver-v0.31.0-linux64.tar.gz \
+  && chmod +x geckodriver \
+  && mv geckodriver /usr/local/bin/
 
-# Copy files generated in the builder image
-COPY --from=builder /app /app
-COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
+CMD RAILS_ENV=${RAILS_ENV} bundle exec rake db:migrate && bundle exec rails s -e ${RAILS_ENV} -p ${PORT} --binding=0.0.0.0
 
-ARG GIT_COMMIT_SHA="UNKNOWN"
-ENV GIT_COMMIT_SHA=${GIT_COMMIT_SHA}
-RUN echo ${GIT_COMMIT_SHA} > ./GIT_COMMIT_SHA
+FROM base as staging
 
-ENV PORT=8080
+CMD RAILS_ENV=${RAILS_ENV} bundle exec rake db:migrate && bundle exec rails s -e ${RAILS_ENV} -p ${PORT} --binding=0.0.0.0
```

Comment:
This may be on your list already, but just in case: I don't think these are going to want to be running the migrations on start up - once we have ecs services trying to start multiple containers at once that will be problematic. 

(It does make sense to do so for local dev, of course...)

Reply:
We've removed these commands as we'll want to run them in the pipeline rather than on the AWS ECS instead anyway (for all envs)